test/runtests.jl

```diff
@@ -72,13 +70,35 @@ using Test
             f(a; kw=1) = a - kw
             a = 2.0
             kw = 1
-            call1 = @cached_call f(a; kw=1)
+            call1 = @cached_call f(a; kw=kw)
             @static if VERSION >= v"1.5"
                 call2 = @cached_call f(a; kw)
-                @test f(a; kw=1) == call1 == call2
+                @test f(a; kw=kw) == call1 == call2
             else
-                @test f(a; kw=1) == call1
+                @test f(a; kw=kw) == call1
             end
         end
+
+        @testset "dot access" begin
+            f(a; kw=0) = a + kw
+            nt = (one=1, two=2)
```

Comment:
are you thinking the use case being only a NamedTuple? Does Dictionary work here?

Reply:
Dictionary also works, though it would be in the other category (square brackets access). Since these are expressions it doesn't matter what the container is - in fact the macro leaves that part intact and it evaluates in the scope of the call.